packages/hardhat-core/src/internal/core/runtime-environment.ts

```diff
@@ -124,7 +126,23 @@ export class Environment implements HardhatRuntimeEnvironment {
       taskArguments
     );
 
-    return this._runTaskDefinition(taskDefinition, resolvedTaskArguments);
+    try {
+      return await this._runTaskDefinition(
```

Comment:
I thought this was going to be placed here: https://github.com/nomiclabs/buidler/blob/2.0/packages/hardhat-core/src/internal/cli/cli.ts#L161

Putting it in the CLI is less complete, but covers the bast majority of cases, and imo is cleaner.

Reply:
Oh, but it makes it MUCH harder to test, so I regret my previous comment.